plane/src/typed_unix_socket/mod.rs

```diff
@@ -65,7 +52,10 @@ mod tests {
             let mut request_rx = server.subscribe_requests();
             while let Ok(request) = request_rx.recv().await {
                 let response = "Hello, client!".to_string();
-                server.send_response(request, response).await.unwrap();
+                server
+                    .send_response(request.id.unwrap(), response)
```

Comment:
(This function call will be cleaner once we change `send_response` to simply take `(request, response)`)

Reply:
My general preference is for functions to take just the data they need to do what they need to do. I find that this makes them easier to test and more versatile when refactoring, and the actual data flow is more accurately represented by the function signatures.

I'm inclined to keep this to ID, but use a newtype as you suggested to take a `MessageId` or similar as the first param instead of a `String`.

Reply:
Just I read your rationale below and I see that side of it too. I do like the idea of hiding the implementation detail of the ID, so I'm fine with that. It does mean that if we want to use the contents of the original message we will need to clone it though, since we won't be able to pass the request to `send_response` if we have already taken the message field from it.

Reply:
Sounds good, I'll pick one of these options and finalize it in the parent PR!